File: Semester3/Advanced Methods of Programming/Toy Language Interpreter C#/Assignement7/Assignement7/Model/RunCommand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Assignement7.Controller_folder;

namespace Assignement7.Model
{
    public class RunCommand : Command
    {
        private Controller ctr;

        public RunCommand(string key, string desciption, Controller ctr) : base(key, desciption)
        {
            this.ctr = ctr;
        }

        public override void Execute()
        {
            try
            {
                ctr.AllSteps();
            }
            catch (Exception exception)
            {
                Console.WriteLine(exception.Message);
            }
        }
    }
}

File: Semester3/Advanced Methods of Programming/Toy Language Interpreter JavaFx/Assignment8/src/View/RunCommand.java
package View;

import Controller.Controller;
import Model.Exceptions.ToyLanguageInterpreterException;

import java.io.FileNotFoundException;
import java.io.IOException;

public class RunCommand extends Command{
    private Controller controller;

    public RunCommand(String key, String description, Controller controller){
        super(key, description);
        this.controller = controller;
    }

    @Override
    public void execute() {
        try{
            controller.allSteps();
        } catch (Exception e) {
            System.out.println(e.getMessage());
        }
    }
}
